selectel/provider.go

```diff
@@ -46,27 +47,63 @@ func Provider() *schema.Provider {
 		Schema: map[string]*schema.Schema{
 			"token": {
 				Type:        schema.TypeString,
-				Required:    true,
+				Optional:    true,
 				DefaultFunc: schema.EnvDefaultFunc("SEL_TOKEN", nil),
 				Description: "Token to authorize with the Selectel API.",
 			},
+			"user": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				RequiredWith: []string{"password", "domain_name"},
+				DefaultFunc:  schema.EnvDefaultFunc("SEL_USER", nil),
+				Description:  "Cloud user.",
+			},
+			"password": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Sensitive:    true,
+				RequiredWith: []string{"user", "domain_name"},
+				DefaultFunc:  schema.EnvDefaultFunc("SEL_PASSWORD", nil),
+				Description:  "Cloud user password.",
+			},
+			"domain_name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				RequiredWith: []string{"user", "password"},
+				DefaultFunc:  schema.EnvDefaultFunc("SEL_DOMAIN_NAME", nil),
+				Description:  "Cloud domain ID to import resources that need the project scope auth token.",
+			},
```

Comment:
Perhaps we should use the same variables(and descriptions?) as in terraform-provider-openstack?